commit fc596a9fe19f1b79f9265f37727e2dacde63647d
Author: Peter Michael Green <plugwash@debian.org>
Date:   Sat Oct 30 17:50:00 2021 +0000

    Fix alloc_slice_fill_zero on architectures where "max alignment" is only 4 (such as i386)

diff --git a/tests/alloc_fill.rs b/tests/alloc_fill.rs
index 70bd84f31..e71eff253 100644
--- a/tests/alloc_fill.rs
+++ b/tests/alloc_fill.rs
@@ -1,5 +1,7 @@
 use bumpalo::Bump;
 use std::alloc::Layout;
+use std::mem;
+use std::cmp;
 
 #[test]
 fn alloc_slice_fill_zero() {
@@ -17,7 +19,8 @@ fn alloc_slice_fill_zero() {
     b.alloc_slice_fill_clone(0, &"hello".to_string());
     b.alloc_slice_fill_default::<String>(0);
     let ptr2 = b.alloc(MyZeroSizedType);
-    assert_eq!(ptr1.as_ptr() as usize & !7, ptr2 as *mut _ as usize);
+    let alignment = cmp::max(mem::align_of::<u64>(),mem::align_of::<String>());
+    assert_eq!(ptr1.as_ptr() as usize & !(alignment - 1), ptr2 as *mut _ as usize);
 
     let ptr3 = b.alloc_layout(layout);
     assert_eq!(ptr2 as *mut _ as usize, ptr3.as_ptr() as usize + 1);
